src/structures/Message.js

```diff
@@ -370,8 +370,8 @@ class Message extends Base {
    *   .catch(console.error);
    */
   awaitReactions(filter, options = {}) {
+    const collector = this.createReactionCollector(filter, options);
     return new Promise((resolve, reject) => {
-      const collector = this.createReactionCollector(filter, options);
```

Comment:
i believe the promise will actually be rejected if an error is thrown inside the callback (so long as said callback function is not async), i've given this a test and that does seem to be the case, so it can be kept inside here